### PR TITLE
Add Qos API [13837]

### PR DESCRIPTION
**Pull request by richiware**
Support for DDS QoS no Python bindings.

Need to be merge first eprosima/fast-dds#2492

### CODE REVIEW

File: fastdds_python/test/api/test_qos.py [hunk 1]
```diff
@@ -0,0 +1,1409 @@
...
+    assert(500 == datareader_qos.reader_resource_limits().outstanding_reads_allocation.maximum)
+    assert(6 == datareader_qos.reader_resource_limits().outstanding_reads_allocation.increment)
+    assert(33 == datareader_qos.reader_resource_limits().max_samples_per_read)
+
+    # .data_sharing
+    datareader_qos.data_sharing().on("/")
+    assert(fastdds.ON == datareader_qos.data_sharing().kind())
+    assert("/" == datareader_qos.data_sharing().shm_directory())
+
+    # Check agains default_topic_qos
```
Comment (jsantiago-eProsima) [inline fastdds_python/test/api/test_qos.py:195]:
```suggestion
    # Check against default_topic_qos
```

File: fastdds_python/test/api/test_qos.py [hunk 2]
```diff
@@ -0,0 +1,1409 @@
...
+    datareader_qos.liveliness().announcement_period.seconds = 30
+    datareader_qos.liveliness().announcement_period.nanosec = 50
+    assert(fastdds.MANUAL_BY_PARTICIPANT_LIVELINESS_QOS == datareader_qos.liveliness().kind)
+    assert(40 == datareader_qos.liveliness().lease_duration.seconds)
+    assert(61 == datareader_qos.liveliness().lease_duration.nanosec)
+    assert(30 == datareader_qos.liveliness().announcement_period.seconds)
+    assert(50 == datareader_qos.liveliness().announcement_period.nanosec)
+
+    # .reliability
+    datareader_qos.reliability().kind = fastdds.RELIABLE_RELIABILITY_QOS
```
Comment (jsantiago-eProsima) [inline fastdds_python/test/api/test_qos.py:39]: Shouldn't we test max_blocking_time?

File: fastdds_python/test/api/test_qos.py [hunk 3]
```diff
@@ -0,0 +1,1409 @@
...
+    assert(10 == datareader_qos.lifespan().duration.seconds)
+    assert(33 == datareader_qos.lifespan().duration.nanosec)
+
+    # .durability_service
+    datareader_qos.durability_service().history_kind = fastdds.KEEP_ALL_HISTORY_QOS
+    datareader_qos.durability_service().history_depth = 10
+    datareader_qos.durability_service().max_samples = 5
+    datareader_qos.durability_service().max_instances = 20
+    datareader_qos.durability_service().max_samples_per_instance = 30
+    assert(fastdds.KEEP_ALL_HISTORY_QOS == datareader_qos.durability_service().history_kind)
```
Comment (jsantiago-eProsima) [inline fastdds_python/test/api/test_qos.py:113]: Should we test service_cleanup_delay?

File: fastdds_python/test/api/test_qos.py [hunk 4]
```diff
@@ -0,0 +1,1409 @@
...
+    assert(10 == datareader_qos.durability_service().history_depth)
+    assert(5 == datareader_qos.durability_service().max_samples)
+    assert(20 == datareader_qos.durability_service().max_instances)
+    assert(30 == datareader_qos.durability_service().max_samples_per_instance)
+
+    # .reliable_reader_qos
+    datareader_qos.reliable_reader_qos().times.initialAcknackDelay.seconds = 34
+    datareader_qos.reliable_reader_qos().times.initialAcknackDelay.nanosec = 32
+    datareader_qos.reliable_reader_qos().times.heartbeatResponseDelay.seconds = 432
+    datareader_qos.reliable_reader_qos().times.heartbeatResponseDelay.nanosec = 43
```
Comment (jsantiago-eProsima) [inline fastdds_python/test/api/test_qos.py:123]: I don't see any mention of the `times` member in the Python API.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
Partial review. Please check my comments regarding test_qos.py missing methods.

File: fastdds_python/CMakeLists.txt [hunk 5]
```diff
@@ -34,6 +34,8 @@ FIND_PACKAGE(SWIG REQUIRED)
 INCLUDE(${SWIG_USE_FILE})
 SET(CMAKE_SWIG_FLAGS "")
 
+find_package (PythonInterp REQUIRED)
+
 FIND_PACKAGE(PythonLibs REQUIRED)
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastdds_python/CMakeLists.txt:39]: Ok, I wil change it

File: fastdds_python/test/api/test_qos.py [hunk 6]
```diff
@@ -0,0 +1,1409 @@
...
+    assert(40 == datareader_qos.reader_resource_limits().sample_infos_allocation.initial)
+    assert(400 == datareader_qos.reader_resource_limits().sample_infos_allocation.maximum)
+    assert(5 == datareader_qos.reader_resource_limits().sample_infos_allocation.increment)
+    assert(50 == datareader_qos.reader_resource_limits().outstanding_reads_allocation.initial)
+    assert(500 == datareader_qos.reader_resource_limits().outstanding_reads_allocation.maximum)
+    assert(6 == datareader_qos.reader_resource_limits().outstanding_reads_allocation.increment)
+    assert(33 == datareader_qos.reader_resource_limits().max_samples_per_read)
+
+    # .data_sharing
+    datareader_qos.data_sharing().on("/")
```
[the remark this answers is not in the document]
Comment (richiware) [reply fastdds_python/test/api/test_qos.py:191]: Ok, I will add them

File: fastdds_python/test/api/test_qos.py (hunk 2 above)
Comment (richiware) [reply fastdds_python/test/api/test_qos.py:39]: I will add it

File: fastdds_python/test/api/test_qos.py (hunk 3 above)
Comment (richiware) [reply fastdds_python/test/api/test_qos.py:113]: I have to check this

File: fastdds_python/test/api/test_qos.py (hunk 4 above)
Comment (richiware) [reply fastdds_python/test/api/test_qos.py:123]: I will check it

File: fastdds_python/CMakeLists.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (richiware) [reply fastdds_python/CMakeLists.txt]: Done

File: fastdds_python/test/api/test_qos.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (richiware) [reply fastdds_python/test/api/test_qos.py:242]: Done

File: fastdds_python/test/api/test_qos.py (hunk 2 above)
Comment (richiware) [reply fastdds_python/test/api/test_qos.py:42]: Done

File: fastdds_python/test/api/test_qos.py (hunk 3 above)
Comment (richiware) [reply fastdds_python/test/api/test_qos.py]: Done

File: fastdds_python/test/api/test_qos.py (hunk 4 above)
Comment (richiware) [reply fastdds_python/test/api/test_qos.py]: `ReaderTimes` is from *RTPS* API (see [this](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/rtps/attributes/ReaderTimes.html?highlight=ReaderTimes)). Python bindings allow to use the *DDS* API and therefore this API is the documented one. But this API uses some times *RTPS* structures. We can talk about this in a meeting.

File: fastdds_python/test/api/test_qos.py [hunk 7]
```diff
@@ -0,0 +1,1770 @@
...
+    assert(fastdds.AUTO == datareader_qos.data_sharing().kind())
+    assert("" == datareader_qos.data_sharing().shm_directory())
+    assert(0 == datareader_qos.data_sharing().max_domains())
+    datareader_qos.data_sharing().on("/")
+    datareader_qos.data_sharing().set_max_domains(3)
+    assert(fastdds.ON == datareader_qos.data_sharing().kind())
+    assert("/" == datareader_qos.data_sharing().shm_directory())
+    assert(3 == datareader_qos.data_sharing().max_domains())
+
+    # Check against default_topic_qos
```
Comment (jsantiago-eProsima) [inline fastdds_python/test/api/test_qos.py:271]:
```suggestion
    # Check against default_datareader_qos
```

File: fastdds_python/test/api/test_qos.py [hunk 8]
```diff
@@ -0,0 +1,1770 @@
...
+    assert(fastdds.AUTO == datawriter_qos.data_sharing().kind())
+    assert("" == datawriter_qos.data_sharing().shm_directory())
+    assert(0 == datawriter_qos.data_sharing().max_domains())
+    datawriter_qos.data_sharing().on("/")
+    datawriter_qos.data_sharing().set_max_domains(3)
+    assert(fastdds.ON == datawriter_qos.data_sharing().kind())
+    assert("/" == datawriter_qos.data_sharing().shm_directory())
+    assert(3 == datawriter_qos.data_sharing().max_domains())
+
+    # Check agains default_topic_qos
```
Comment (jsantiago-eProsima) [inline fastdds_python/test/api/test_qos.py:687]:
```suggestion
    # Check agains default_datawriter_qos
```

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
Just two small copy/paste comment errors. Also, DomainParticipantFactoryQos is not being tested at the moment. All of the other autoenablement qos are being tested so I think we should test that one too.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
These changes look good. I would like this test file to have a DomainParticipantFactoryQos test similar to [this one](https://github.com/eProsima/Fast-DDS/blob/04abfbac4e6a32ad9efb519d5a3dfb1bad968fbd/test/unittest/dds/participant/ParticipantTests.cpp#L283-L300) at least, since i don't think that one is being tested.

**Pull request merged**